pkg/generic/generic.go

```diff
@@ -76,14 +76,22 @@ func MapThriftGenericForJSON(p DescriptorProvider) (Generic, error) {
 //
 //	g, err := generic.HTTPThriftGeneric(p)
 //	SetBinaryWithBase64(g, true)
-func HTTPThriftGeneric(p DescriptorProvider) (Generic, error) {
-	codec, err := newHTTPThriftCodec(p, thriftCodec)
-	if err != nil {
-		return nil, err
+func HTTPThriftGeneric(p DescriptorProvider, opts ...Option) (Generic, error) {
+	var codec *httpThriftCodec
+	var err error
+	if opts != nil {
```

Comment:
why check opts nil here? `codec, err = newHTTPThriftCodec(p, thriftCodec, opts...)` is ok for nil opts param

Reply:
@YangruiEmma yes, this code is not right. I'm fixing that now.

Reply:
fixed